Sources/SwiftIndexStore/SwiftIndexStore.swift

```diff
@@ -56,12 +60,16 @@ public final class IndexStore {
 
     public func mainFilePath(for unit: IndexStoreUnit) throws -> String? {
         let reader = try createUnitReader(for: unit)
-        return lib.unit_reader_get_main_file(reader).toSwiftString()
+        let mainFile = lib.unit_reader_get_main_file(reader).toSwiftString()
+        lib.unit_reader_dispose(reader)
```

Comment:
How about using `defer` intead?

Reply:
The reader will not be created if creation throw an error

Reply:
That's right for now, but I would like to make it more maintainable for future changes. and also for consistency.

Reply:
ok, fixed it